packages/vite/src/node/optimizer/index.ts

```diff
@@ -620,7 +620,11 @@ export function getOptimizedDepPath(
 function getDepsCacheSuffix(config: ResolvedConfig): string {
   let suffix = ''
   if (config.command === 'build') {
-    suffix += '_build'
+    // Differentiate build caches depending on outDir to allow parallel builds
+    const { outDir } = config.build
+    const buildId =
+      outDir.length > 8 || outDir.includes('/') ? getHash(outDir) : outDir
```

Comment:
Maybe we can use `flattenId` here so that we don't have to potentially hash it? 🤔

Reply:
good point. This makes the path easier to debug 👍

Reply:
good point. This makes the path easier to debug 👍

Reply:
That is an interesting idea. So we could flatten and check that the length isn't too much? But maybe a limit like 16 🤔
My main issue was if there is a chance for the user to set an absolute or relative path out of root. But we could detect that, maybe it isn't even a thing?

Reply:
There is a problem with using `flattenId`. We can safely use it with node deps because we know that `_` isn't a valid char there. But here if we use it we could collide `dist/foo` with `dist_foo`. I agree it would be extremely weird to have both, but I don't feel that comfortable with making this a special case. I think we should leave the code as it is now. What do you think @bluwy?

Reply:
That's true, I guess if we still want to, we can pre-replace `_` with `____`. My main concern is that `getDepsCacheSuffix` is a hot-code path, and potentially calling `getHash` many times may not be good for perf. Unless we refactor the optimizer to pre-calculate the `cacheDir`.

We could probably still use `getHash` if we plan to optimize it later on.

Reply:
Ok, let's merge this now as it is an important fix and leave the optimization for a future PR. Right now the big majority of users won't hit getHash anyways.